### PR TITLE
Fix issue #5: Search bar covering whole header on medium devices

### LINKED ISSUE #5: Search bar covers whole header on medium (and above) devices on refresh
### Issue Description

Search bar covers whole header on medium devices on refreshing the page.

### Expected Behavior

Search bar should not cover whole header section on medium devices

### Actual Behavior

Search bar covers whole header section on medium (and above) devices

### Steps to Reproduce (for Bugs)

Steps are shown in the attached video file

https://github.com/Tauseef-Hilal/FusionFlix/assets/98336906/2c030ac0-4629-497c-9714-2c0c698392d4

### PR DESCRIPTION
**Pull request by Tauseef-Hilal**
### Pull Request Checklist

- [x] **Type of Change:**

  - [x] Bug Fix
  - [ ] Feature Addition
  - [ ] Code Refactoring
  - [ ] Documentation Update
  - [ ] Other (please specify)

### Description
Fix search bar covering whole header on medium (and above) devices on refresh.
+ Optimized rebuilds due to window resize event in Header 

- [x] **Related Issues:**
Fixes #5 

### Reviewer Guidelines

- [x] **Code Quality:**

  - [x] Code follows the project's coding standards.
  - [x] Code is well-documented and easy to understand.

- [x] **Functionality:**

  - [x] The changes achieve the intended functionality.
  - [x] No unintended side effects or regressions have been introduced.

### CODE REVIEW

File: src/components/Header.tsx [hunk 1]
```diff
@@ -17,12 +17,25 @@ const Header: React.FC = () => {
 
   useEffect(() => {
     function resizeHandler() {
+      // Performance optimisation: Only update viewportWidth if it lies on the 
+      // opposite side of window.innerWidth, about 768
+      console.log(viewportWidth, window.innerWidth)
```
Comment (Suhaib-Hilal) [inline src/components/Header.tsx]:
```suggestion

```

File: src/components/Header.tsx [hunk 2]
```diff
@@ -17,12 +17,25 @@ const Header: React.FC = () => {
 
   useEffect(() => {
     function resizeHandler() {
+      // Performance optimisation: Only update viewportWidth if it lies on the 
+      // opposite side of window.innerWidth, about 768
```
Comment (Suhaib-Hilal) [inline src/components/Header.tsx]: Maybe this one's easier to understand:
```
      // Performance optimisation: Only update viewportWidth if window.innerWidth
      // and viewportWidth both are not less/greater than or equal to 768
```

**Review by Suhaib-Hilal — CHANGES_REQUESTED** [review]

File: src/components/Header.tsx (hunk 2 above)
Comment (Tauseef-Hilal) [reply src/components/Header.tsx]: Sounds good

**Review by Suhaib-Hilal — CHANGES_REQUESTED** [review]
Forgot to mention that you should create a constant for 768

**Review by Suhaib-Hilal — APPROVED** [review]

**Pull request merged**